### ISSUE TITLE
IGNF codes cannot be saved properly into gpkg_spatial_ref_sys

### ISSUE THREAD

**Issue by jratike80** [opengeospatial/geopackage#688]
Catched from https://gis.stackexchange.com/questions/486339/transfer-from-qgis-to-arcgis-pro

The structure of the gpkg_spatial_ref_sys table is 

```
CREATE TABLE gpkg_spatial_ref_sys (
  srs_name TEXT NOT NULL,
  srs_id INTEGER NOT NULL PRIMARY KEY,
  organization TEXT NOT NULL,
  organization_coordsys_id INTEGER NOT NULL,
  definition  TEXT NOT NULL,
  description TEXT
);
```

The organization_coordsys_id cannot be NULL and it must be an integer. However IGNF is unfortunately using IDs which are strings. For example `IGNF:ETRS89` https://spatialreference.org/ref/ignf/ETRS89/wkt2.html.
It means that applications must store some fake integer as organization_coordsys_id , and for knowing what the CRS really is, the only possibility is to interpret the definition string.

Alternative solutions:
- IGNF will create integer IDs
- datatype of the organization_coordsys_id column could be changed into `TEXT` (but if the ID is a string that contains non-numeric characters then it could not be re-used as srs_id - that does not matter, srs_id just needs to be unique)
- datatype of the organization_coordsys_id column could be changed into a special non-standard datatype `INTEGER_OR_TEXT` that is fine for SQLite (this is what the Proj library is using in its proj.db database) but some applications may get confused if the id is sometimes string, sometimes integer
- add column "organization coordsys_string_id TEXT" for IGNF and others who use sting IDs (but what to do with organization_coordsys_id , could it still be NOT NULL and filled with fake values?)

**Comment by jyutzler** [comment]
I'm not sure I see the problem here. Why not just make the `organization_coordsys_id` "89" and the `organization` "IGNF:ETRS"? The point is to create uniqueness and I believe this does that.

https://opengeospatial.github.io/e-learning/geopackage/text/contents.html#id2

**Comment by jratike80** [comment]
I have thought that the `organization_coordsys_id` is somehow supposed to be the ID that the organization is using. What integer ID would you use for `IGNF:LNGPGG`

```
GEOGCRS["Lambert Nord de Guerre grades Paris",
    DATUM["Nord de Guerre (Paris)",
        ELLIPSOID["Plessis 1817",6376523,308.64,
            LENGTHUNIT["metre",1]]],
    PRIMEM["Paris",2.5969213,
        ANGLEUNIT["grad",0.0157079632679489]],
    CS[ellipsoidal,2],
        AXIS["geodetic longitude (Lon)",east,
            ORDER[1],
            ANGLEUNIT["grad",0.0157079632679489]],
        AXIS["geodetic latitude (Lat)",north,
            ORDER[2],
            ANGLEUNIT["grad",0.0157079632679489]],
    USAGE[
        SCOPE["LOCALE, HISTORIQUE"],
        AREA["World."],
        BBOX[-90,-180,90,180]],
    ID["IGNF","LNGPGG"]]
```

I can see that Proj has resolved this by using a special datatype `INTEGER_OR_TEXT` in several places in the proj.db database.

<img width="308" alt="image" src="https://github.com/user-attachments/assets/7bf27902-2b6e-4474-b3ea-3a4a407e648a">

<img width="305" alt="image" src="https://github.com/user-attachments/assets/e3b01c61-9596-454f-85f4-2f37f1739af1">

**Comment by jyutzler** [comment]
Eek. I think they will have to invent an integer for that case.

Keep in mind that full interoperability is never going to be possible here. The set of possible CRSs is unbounded and there is no robust way to force implementers to use a consistent set of IDs. I would expect this group to use the `description` column to its maximum to make this as clear as they can and hope that an external agreement (i.e., profile) would provide the necessary interoperability for a particular community of interest. When this fails, capable clients can use the `definition` just like any other CRS they have never heard of.

I put this in the category of things that we would redo if starting from scratch, but that the disruption to the community by making a GeoPackage 2.0 would be so severe that we are just going to have to live with it.

**Comment by jratike80** [comment]
I made some testing and I'll add my findings here for future reference.

GDAL (version 3.10) writes `IGNF:LNGPGG` this way into gpkg_spatial_ref_sys

```
srs_name: Lambert Nord de Guerre grades Paris
srs_id: 100000
organization: NONE
organization_coordsys_id: 100000
definition: GEOGCS["Lambert Nord de Guerre grades Paris",DATUM["Nord_de_Guerre_Paris",SPHEROID["Plessis 1817",6376523,308.64,AUTHORITY["EPSG","7027"]],AUTHORITY["EPSG","6902"]],PRIMEM["Paris",2.33722917,AUTHORITY["EPSG","8903"]],UNIT["grad",0.0157079632679489,AUTHORITY["EPSG","9105"]],AXIS["Longitude",EAST],AXIS["Latitude",NORTH],AUTHORITY["IGNF","LNGPGG"]]"
description: NULL
```
GDAL and QGIS also read the CRS correctly back. As far as I understand it is only possible if they fall back to interpret the WKT from the definition field, because the code `NONE:100000` is meaningless. I think this is a good method that could be recommended also for others.

I tried also to open the GPKG that was written with GDAL by using ArcGIS Pro 2.7.0. It could read the tables with EPSG CRS without issues, but not at all tables having IGNF CRS. The error pop-up from ArcGIS looks like this:



<img width="283" alt="image" src="https://github.com/user-attachments/assets/58ed7ca7-2921-4373-acf9-c9722032e9f8">
